Podspec CocoaPods 1.0 compatibility

## The given:

Creating a new iOS project, linking themis master as CocoaPod.

## The problem:

If you create new project using CocoaPods v1.0.1, your podfile includes `use_frameworks!` and links to the HEAD of master branch, then Pods can't be installed.

#### podfile:

```
source 'https://github.com/CocoaPods/Specs.git'

platform :ios, '9.0'
project 'NoWorkingPodsDemo/NoWorkingPodsDemo.xcodeproj'
inhibit_all_warnings!
use_frameworks!

install! 'cocoapods', :deterministic_uuids => false

target :"NoWorkingPodsDemo" do

   # example should work with head
  pod 'themis', :git => "https://github.com/cossacklabs/themis.git"

end
```

#### Error log

```
Analyzing dependencies
Pre-downloading: `themis` from `https://github.com/cossacklabs/themis.git`
^[[A^[[A[!] Unable to find a specification for 'themis'.

[!] Unable to load a podspec from `themis.podspec`, skipping:

Pod::DSLError
```

Last issue may be fixed by disabling modules, as described on [apple forums](https://discussions.apple.com/message/24766902#24766902).

So, resulting wrapper contains built-in openssl frameworks and disabling modules flag.

### Podspec
Link to frameworks.
Drop iOS 7.
Disable modules.

```
Pod::Spec.new do |s|
    s.name = "themis"
    s.version = "0.9.3"
    s.summary = "Data security library for network communication and data storage for iOS and OS X "
    s.description = "Themis is a data security library, providing users with high-quality security services for secure messaging of any kinds and flexible data storage. Themis is aimed at modern development practices, with high level OOP wrappers for Ruby, Python, PHP, Java / Android and iOS / OSX. It is designed with ease of use in mind, high security and cross-platform availability."
    s.homepage = "http://cossacklabs.com"
    s.license = { :type => 'Apache 2.0'}
    s.source = { :git => "https://github.com/cossacklabs/themis.git", :branch => "pods_1_0" }
    s.author = {'cossacklabs' => 'info@cossacklabs.com'}
    
    s.ios.vendored_frameworks  = 'src/wrappers/themis/Obj-C/openssl_frameworks/ios/openssl.framework'
    s.osx.vendored_frameworks  = 'src/wrappers/themis/Obj-C/openssl_frameworks/macos/openssl.framework'

    s.ios.deployment_target = '8.0'
    s.osx.deployment_target = '10.9'
    s.requires_arc = false

    s.xcconfig = { 'OTHER_CFLAGS' => '-DLIBRESSL', 'USE_HEADERMAP' => 'NO', 
        'HEADER_SEARCH_PATHS' => '"${PODS_ROOT}/themis/src" "${PODS_ROOT}/themis/src/wrappers/themis/Obj-C"',
        'CLANG_ENABLE_MODULES' => 'NO' }


    s.subspec 'core' do |ss|
        ss.source_files = "src/themis/*.{h,c}", "src/soter/**/*.{c,h}"
        ss.header_mappings_dir = "src"
        ss.header_dir = 'src'
        ss.preserve_paths = "src/themis/*.h", "src/soter/**/*.h"
        ss.public_header_files = "src/themis/*.h", "src/soter/**/*.h"
    end
    
    s.subspec 'objcwrapper' do |ss|
        ss.header_mappings_dir = 'src/wrappers/themis/Obj-C/objcthemis'
        ss.source_files = "src/wrappers/themis/Obj-C/objcthemis/*.{m,h}"
        ss.public_header_files = 'src/wrappers/themis/Obj-C/objcthemis/*.h'
        ss.header_dir = 'objcthemis'
        ss.dependency 'themis/core'
    end
end
```

Project is now building successfully, but you may need to use Themis via Bridging header.

```
#ifndef DemoWithFrameworks_Bridging_Header_h
#define DemoWithFrameworks_Bridging_Header_h

    #import <objcthemis/objcthemis.h>

#endif /* DemoWithFrameworks_Bridging_Header_h */
```

### Linting

However, linting on CP 1.0 fails due to 

```
    - NOTE  | [themis/core,themis/objcwrapper] xcodebuild:  themis/src/soter/openssl/soter_openssl.h:21:10: fatal error: 'openssl/evp.h' file not found
    - NOTE  | [OSX] [themis/objcwrapper] xcodebuild:  error: unable to read module map contents from 'Target Support Files/themis/themis.modulemap': Error Domain=NSCocoaErrorDomain Code=260 "The file “themis.modulemap” couldn’t be opened because there is no such file." UserInfo={NSFilePath=Target Support Files/themis/themis.modulemap, NSUnderlyingError=0x7f902b00a580 {Error Domain=NSPOSIXErrorDomain Code=2 "No such file or directory"}}
    - NOTE  | [OSX] [themis/objcwrapper] xcodebuild:  error: unable to read module map contents from 'Target Support Files/themis/themis.modulemap': Error Domain=NSCocoaErrorDomain Code=260 "The file “themis.modulemap” couldn’t be opened because there is no such file." UserInfo={NSFilePath=Target Support Files/themis/themis.modulemap, NSUnderlyingError=0x7f902afb4540 {Error Domain=NSPOSIXErrorDomain Code=2 "No such file or directory"}}
```

¯\_(ツ)_/¯

--- a/docs/examples/Themis-server/Obj-C/WorkingWithThemisServer/WorkingWithThemisServer.xcodeproj/project.pbxproj
+++ b/docs/examples/Themis-server/Obj-C/WorkingWithThemisServer/WorkingWithThemisServer.xcodeproj/project.pbxproj
@@ -181,12 +181,12 @@
 			isa = PBXNativeTarget;
 			buildConfigurationList = 733BD9E41BDBF58600C6DA2F /* Build configuration list for PBXNativeTarget "WorkingWithThemisServer" */;
 			buildPhases = (
-				794624F656ED293B3C272CF1 /* Check Pods Manifest.lock */,
+				794624F656ED293B3C272CF1 /* [CP] Check Pods Manifest.lock */,
 				733BD9B31BDBF58500C6DA2F /* Sources */,
 				733BD9B41BDBF58500C6DA2F /* Frameworks */,
 				733BD9B51BDBF58500C6DA2F /* Resources */,
-				013706136CE1D47ABDFEC437 /* Copy Pods Resources */,
-				FE9FD3CD2E89491EF5A06321 /* Embed Pods Frameworks */,
+				013706136CE1D47ABDFEC437 /* [CP] Copy Pods Resources */,
+				FE9FD3CD2E89491EF5A06321 /* [CP] Embed Pods Frameworks */,
 			);
 			buildRules = (
 			);
@@ -303,14 +303,14 @@
 /* End PBXResourcesBuildPhase section */
 
 /* Begin PBXShellScriptBuildPhase section */
-		013706136CE1D47ABDFEC437 /* Copy Pods Resources */ = {
+		013706136CE1D47ABDFEC437 /* [CP] Copy Pods Resources */ = {
 			isa = PBXShellScriptBuildPhase;
 			buildActionMask = 2147483647;
 			files = (
 			);
 			inputPaths = (
 			);
-			name = "Copy Pods Resources";
+			name = "[CP] Copy Pods Resources";
 			outputPaths = (
 			);
 			runOnlyForDeploymentPostprocessing = 0;
@@ -318,14 +318,14 @@
 			shellScript = "\"${SRCROOT}/../Pods/Target Support Files/Pods-WorkingWithThemisServer/Pods-WorkingWithThemisServer-resources.sh\"\n";
 			showEnvVarsInLog = 0;
 		};
-		794624F656ED293B3C272CF1 /* Check Pods Manifest.lock */ = {
+		794624F656ED293B3C272CF1 /* [CP] Check Pods Manifest.lock */ = {
 			isa = PBXShellScriptBuildPhase;
 			buildActionMask = 2147483647;
 			files = (
 			);
 			inputPaths = (
 			);
-			name = "Check Pods Manifest.lock";
+			name = "[CP] Check Pods Manifest.lock";
 			outputPaths = (
 			);
 			runOnlyForDeploymentPostprocessing = 0;
@@ -333,14 +333,14 @@
 			shellScript = "diff \"${PODS_ROOT}/../Podfile.lock\" \"${PODS_ROOT}/Manifest.lock\" > /dev/null\nif [[ $? != 0 ]] ; then\n    cat << EOM\nerror: The sandbox is not in sync with the Podfile.lock. Run 'pod install' or update your CocoaPods installation.\nEOM\n    exit 1\nfi\n";
 			showEnvVarsInLog = 0;
 		};
-		FE9FD3CD2E89491EF5A06321 /* Embed Pods Frameworks */ = {
+		FE9FD3CD2E89491EF5A06321 /* [CP] Embed Pods Frameworks */ = {
 			isa = PBXShellScriptBuildPhase;
 			buildActionMask = 2147483647;
 			files = (
 			);
 			inputPaths = (
 			);
-			name = "Embed Pods Frameworks";
+			name = "[CP] Embed Pods Frameworks";
 			outputPaths = (
 			);
 			runOnlyForDeploymentPostprocessing = 0;

--- a/docs/examples/Themis-server/swift/SwiftThemisServerExample/SwiftThemisServerExample.xcodeproj/project.pbxproj
+++ b/docs/examples/Themis-server/swift/SwiftThemisServerExample/SwiftThemisServerExample.xcodeproj/project.pbxproj
@@ -103,12 +103,12 @@
 			isa = PBXNativeTarget;
 			buildConfigurationList = 736DDB741CC5990900CD0C31 /* Build configuration list for PBXNativeTarget "SwiftThemisServerExample" */;
 			buildPhases = (
-				7521C531C10F5BBDA84E6229 /* Check Pods Manifest.lock */,
+				7521C531C10F5BBDA84E6229 /* [CP] Check Pods Manifest.lock */,
 				736DDB5E1CC5990900CD0C31 /* Sources */,
 				736DDB5F1CC5990900CD0C31 /* Frameworks */,
 				736DDB601CC5990900CD0C31 /* Resources */,
-				34970ACCC23BE6291712C3E6 /* Embed Pods Frameworks */,
-				218482DD932A2BEB2D27AEC9 /* Copy Pods Resources */,
+				34970ACCC23BE6291712C3E6 /* [CP] Embed Pods Frameworks */,
+				218482DD932A2BEB2D27AEC9 /* [CP] Copy Pods Resources */,
 			);
 			buildRules = (
 			);
@@ -166,14 +166,14 @@
 /* End PBXResourcesBuildPhase section */
 
 /* Begin PBXShellScriptBuildPhase section */
-		218482DD932A2BEB2D27AEC9 /* Copy Pods Resources */ = {
+		218482DD932A2BEB2D27AEC9 /* [CP] Copy Pods Resources */ = {
 			isa = PBXShellScriptBuildPhase;
 			buildActionMask = 2147483647;
 			files = (
 			);
 			inputPaths = (
 			);
-			name = "Copy Pods Resources";
+			name = "[CP] Copy Pods Resources";
 			outputPaths = (
 			);
 			runOnlyForDeploymentPostprocessing = 0;
@@ -181,14 +181,14 @@
 			shellScript = "\"${SRCROOT}/../Pods/Target Support Files/Pods-SwiftThemisServerExample/Pods-SwiftThemisServerExample-resources.sh\"\n";
 			showEnvVarsInLog = 0;
 		};
-		34970ACCC23BE6291712C3E6 /* Embed Pods Frameworks */ = {
+		34970ACCC23BE6291712C3E6 /* [CP] Embed Pods Frameworks */ = {
 			isa = PBXShellScriptBuildPhase;
 			buildActionMask = 2147483647;
 			files = (
 			);
 			inputPaths = (
 			);
-			name = "Embed Pods Frameworks";
+			name = "[CP] Embed Pods Frameworks";
 			outputPaths = (
 			);
 			runOnlyForDeploymentPostprocessing = 0;
@@ -196,14 +196,14 @@
 			shellScript = "\"${SRCROOT}/../Pods/Target Support Files/Pods-SwiftThemisServerExample/Pods-SwiftThemisServerExample-frameworks.sh\"\n";
 			showEnvVarsInLog = 0;
 		};
-		7521C531C10F5BBDA84E6229 /* Check Pods Manifest.lock */ = {
+		7521C531C10F5BBDA84E6229 /* [CP] Check Pods Manifest.lock */ = {
 			isa = PBXShellScriptBuildPhase;
 			buildActionMask = 2147483647;
 			files = (
 			);
 			inputPaths = (
 			);
-			name = "Check Pods Manifest.lock";
+			name = "[CP] Check Pods Manifest.lock";
 			outputPaths = (
 			);
 			runOnlyForDeploymentPostprocessing = 0;

--- a/docs/examples/objc/ThemisTest/ThemisTest.xcodeproj/project.pbxproj
+++ b/docs/examples/objc/ThemisTest/ThemisTest.xcodeproj/project.pbxproj
@@ -188,12 +188,12 @@
 			isa = PBXNativeTarget;
 			buildConfigurationList = 731624B51ADB174200972A7A /* Build configuration list for PBXNativeTarget "ThemisTest" */;
 			buildPhases = (
-				752A804A4446800FFB3ADCD7 /* Check Pods Manifest.lock */,
+				752A804A4446800FFB3ADCD7 /* [CP] Check Pods Manifest.lock */,
 				7316248E1ADB174200972A7A /* Sources */,
 				7316248F1ADB174200972A7A /* Frameworks */,
 				731624901ADB174200972A7A /* Resources */,
-				9A078C251A561873552D1DC1 /* Copy Pods Resources */,
-				D4BA3E32BDAA4C42E16EDE5D /* Embed Pods Frameworks */,
+				9A078C251A561873552D1DC1 /* [CP] Copy Pods Resources */,
+				D4BA3E32BDAA4C42E16EDE5D /* [CP] Embed Pods Frameworks */,
 			);
 			buildRules = (
 			);
@@ -284,14 +284,14 @@
 /* End PBXResourcesBuildPhase section */
 
 /* Begin PBXShellScriptBuildPhase section */
-		752A804A4446800FFB3ADCD7 /* Check Pods Manifest.lock */ = {
+		752A804A4446800FFB3ADCD7 /* [CP] Check Pods Manifest.lock */ = {
 			isa = PBXShellScriptBuildPhase;
 			buildActionMask = 2147483647;
 			files = (
 			);
 			inputPaths = (
 			);
-			name = "Check Pods Manifest.lock";
+			name = "[CP] Check Pods Manifest.lock";
 			outputPaths = (
 			);
 			runOnlyForDeploymentPostprocessing = 0;
@@ -299,14 +299,14 @@
 			shellScript = "diff \"${PODS_ROOT}/../Podfile.lock\" \"${PODS_ROOT}/Manifest.lock\" > /dev/null\nif [[ $? != 0 ]] ; then\n    cat << EOM\nerror: The sandbox is not in sync with the Podfile.lock. Run 'pod install' or update your CocoaPods installation.\nEOM\n    exit 1\nfi\n";
 			showEnvVarsInLog = 0;
 		};
-		9A078C251A561873552D1DC1 /* Copy Pods Resources */ = {
+		9A078C251A561873552D1DC1 /* [CP] Copy Pods Resources */ = {
 			isa = PBXShellScriptBuildPhase;
 			buildActionMask = 2147483647;
 			files = (
 			);
 			inputPaths = (
 			);
-			name = "Copy Pods Resources";
+			name = "[CP] Copy Pods Resources";
 			outputPaths = (
 			);
 			runOnlyForDeploymentPostprocessing = 0;
@@ -314,14 +314,14 @@
 			shellScript = "\"${SRCROOT}/../Pods/Target Support Files/Pods-ThemisTest/Pods-ThemisTest-resources.sh\"\n";
 			showEnvVarsInLog = 0;
 		};
-		D4BA3E32BDAA4C42E16EDE5D /* Embed Pods Frameworks */ = {
+		D4BA3E32BDAA4C42E16EDE5D /* [CP] Embed Pods Frameworks */ = {
 			isa = PBXShellScriptBuildPhase;
 			buildActionMask = 2147483647;
 			files = (
 			);
 			inputPaths = (
 			);
-			name = "Embed Pods Frameworks";
+			name = "[CP] Embed Pods Frameworks";
 			outputPaths = (
 			);
 			runOnlyForDeploymentPostprocessing = 0;

--- a/docs/examples/swift/ThemisSwift/ThemisSwift.xcodeproj/project.pbxproj
+++ b/docs/examples/swift/ThemisSwift/ThemisSwift.xcodeproj/project.pbxproj
@@ -141,12 +141,12 @@
 			isa = PBXNativeTarget;
 			buildConfigurationList = 739981291CC42C880095138F /* Build configuration list for PBXNativeTarget "ThemisSwift" */;
 			buildPhases = (
-				639E2197E7C5E165A9FDB72A /* Check Pods Manifest.lock */,
+				639E2197E7C5E165A9FDB72A /* [CP] Check Pods Manifest.lock */,
 				739981131CC42C880095138F /* Sources */,
 				739981141CC42C880095138F /* Frameworks */,
 				739981151CC42C880095138F /* Resources */,
-				BF1CE387B5ACF2A2CC59FB5F /* Embed Pods Frameworks */,
-				DC3FF580972B0C60E16EC322 /* Copy Pods Resources */,
+				BF1CE387B5ACF2A2CC59FB5F /* [CP] Embed Pods Frameworks */,
+				DC3FF580972B0C60E16EC322 /* [CP] Copy Pods Resources */,
 				73688AFE1CC4504F00D3C430 /* ShellScript */,
 			);
 			buildRules = (
@@ -209,14 +209,14 @@
 /* End PBXResourcesBuildPhase section */
 
 /* Begin PBXShellScriptBuildPhase section */
-		639E2197E7C5E165A9FDB72A /* Check Pods Manifest.lock */ = {
+		639E2197E7C5E165A9FDB72A /* [CP] Check Pods Manifest.lock */ = {
 			isa = PBXShellScriptBuildPhase;
 			buildActionMask = 2147483647;
 			files = (
 			);
 			inputPaths = (
 			);
-			name = "Check Pods Manifest.lock";
+			name = "[CP] Check Pods Manifest.lock";
 			outputPaths = (
 			);
 			runOnlyForDeploymentPostprocessing = 0;
@@ -237,14 +237,14 @@
 			shellPath = /bin/sh;
 			shellScript = "if which swiftlint >/dev/null; then\n# uncomment for autocorrection\n# swiftlint autocorrect\n    swiftlint lint --quiet\nelse\n    echo \"SwiftLint does not exist, download from https://github.com/realm/SwiftLint\"\nfi";
 		};
-		BF1CE387B5ACF2A2CC59FB5F /* Embed Pods Frameworks */ = {
+		BF1CE387B5ACF2A2CC59FB5F /* [CP] Embed Pods Frameworks */ = {
 			isa = PBXShellScriptBuildPhase;
 			buildActionMask = 2147483647;
 			files = (
 			);
 			inputPaths = (
 			);
-			name = "Embed Pods Frameworks";
+			name = "[CP] Embed Pods Frameworks";
 			outputPaths = (
 			);
 			runOnlyForDeploymentPostprocessing = 0;
@@ -252,14 +252,14 @@
 			shellScript = "\"${SRCROOT}/../Pods/Target Support Files/Pods-ThemisSwift/Pods-ThemisSwift-frameworks.sh\"\n";
 			showEnvVarsInLog = 0;
 		};
-		DC3FF580972B0C60E16EC322 /* Copy Pods Resources */ = {
+		DC3FF580972B0C60E16EC322 /* [CP] Copy Pods Resources */ = {
 			isa = PBXShellScriptBuildPhase;
 			buildActionMask = 2147483647;
 			files = (
 			);
 			inputPaths = (
 			);
-			name = "Copy Pods Resources";
+			name = "[CP] Copy Pods Resources";
 			outputPaths = (
 			);
 			runOnlyForDeploymentPostprocessing = 0;

--- a/themis.podspec
+++ b/themis.podspec
@@ -7,20 +7,24 @@ Pod::Spec.new do |s|
     s.license = { :type => 'Apache 2.0'}
     s.source = { :git => "https://github.com/cossacklabs/themis.git", :tag => "0.9.3" }
     s.author = {'cossacklabs' => 'info@cossacklabs.com'}
-    s.dependency 'OpenSSL-Universal', '1.0.1.p'
+    
+    s.dependency 'OpenSSL-Universal', '1.0.1.19'
 
-    s.ios.platform = :ios, '7.0'
     s.ios.deployment_target = '7.0'
-
-    s.osx.platform = :osx, '10.9'
     s.osx.deployment_target = '10.9'
     
-    s.xcconfig = { 'OTHER_CFLAGS' => '-DLIBRESSL', 'USE_HEADERMAP' => 'NO'} 
-
+    s.xcconfig = { 'OTHER_CFLAGS' => '-DLIBRESSL', 'USE_HEADERMAP' => 'NO', 
+        'HEADER_SEARCH_PATHS' => '"${PODS_ROOT}/themis/src" "${PODS_ROOT}/themis/src/wrappers/themis/Obj-C"',
+        'LIBRARY_SEARCH_PATHS' => '"${PODS_ROOT}/OpenSSL-Universal/lib-ios"' }
+        
+    # open ssl
+    s.libraries = 'ssl', 'crypto'
     
     s.subspec 'core' do |ss|
         ss.source_files = "src/themis/*.{h,c}", "src/soter/**/*.{c,h}"
         ss.header_mappings_dir = "src"
+        ss.header_dir = 'src'
+        ss.preserve_paths = "src/themis/*.h", "src/soter/**/*.h"
         ss.public_header_files = "src/themis/*.h", "src/soter/**/*.h"
     end
     

